client/ayon_core/hosts/substancepainter/plugins/create/create_textures.py

```diff
@@ -28,9 +28,14 @@ class CreateTextures(Creator):
     icon = "picture-o"
 
     default_variant = "Main"
+    channel_mapping = []
 
-    def create(self, product_name, instance_data, pre_create_data):
+    def apply_settings(self, project_settings):
+        settings = project_settings["substancepainter"].get("create")  # noqa
+        self.channel_mapping = settings["CreateTextures"].get("channel_mapping")
```

Comment:
This `.get()` will return `None` if not found which will error on `get_instance_attr_defs`.
This should like be `get("channel_mapping", [])`?

Reply:
i am thinking about supporting the backward compatibility for this as it will error out the creator if there is no enum value for channel_mapping